rxjava-support/src/main/java/tech/picnic/rx/PicnicRxPlugins.java

```diff
@@ -46,13 +46,13 @@ public static void configureContextPropagation(Collection<RxThreadLocal<?>> rxTh
 
   private static Function<Runnable, Runnable> compose(
       Collection<? extends Function<? super Runnable, ? extends Runnable>> functions) {
-    @SuppressWarnings("unchecked")
-    Stream<Function<Runnable, Runnable>> stream =
-        (Stream<Function<Runnable, Runnable>>) functions.stream();
-    return stream
-        .reduce((r1, r2) -> r -> r2.apply(r1.apply(r)))
-        .orElseThrow(
-            () -> new IllegalArgumentException("At least one `RxThreadLocal` must be provided"));
+    return r -> {
+      @Var Runnable runnable = r;
+      for (Function<? super Runnable, ? extends Runnable> f : functions) {
+        runnable = f.apply(runnable);
```

Comment:
This way later functions "wrap" (have a larger scope) than those listed earlier. Perhaps `runnable.apply(f)` would yield a more intuitive nesting?

Reply:
Eh, `runnable.apply(f)` makes no sense. What I meant to say: maybe we should apply them in reverse order?